examples/newrelic/README.md

```diff
@@ -0,0 +1,193 @@
+
+If you are using a released version of Kubernetes, you should
+refer to the docs that go with that version.
+
+<strong>
+The latest 1.0.x release of this document can be found
+[here](http://releases.k8s.io/release-1.0/examples/newrelic/README.md).
+
+Documentation for other releases can be found at
+[releases.k8s.io](http://releases.k8s.io).
+</strong>
+--
+
+<!-- END STRIP_FOR_RELEASE -->
+
+<!-- END MUNGE: UNVERSIONED_WARNING -->
+
+## NewRelic Server Monitor Example
+
+This example shows how to run a NewRelic server monitoring daemon as a pod in a DaemonSet on an existing Kubernetes cluster.
+
+This example will create a DaemonSet which places the NewRelic monitoring daemon on every node in the cluster. It's also fairly trivial to exclude specific Kubernetes nodes from the DaemonSet to just monitor specific servers.
+
+### Step 0: Prerequisites
+
+This process will create priviliged containers which have full access to the host system for logging. Beware of the security implications of this.
+
+If you are using a Salt based KUBERNETES\_PROVIDER (**gce**, **vagrant**, **aws**), you should enable the
+ability to create privileged containers via the API.
+
+```sh
+$ cd kubernetes
+$ vi cluster/saltbase/pillar/privilege.sls
+
+# If true, allow privileged containers to be created by API
+allow_privileged: true
+```
+
+### Step 1: Configure NewRelic
+
+NewRelic is configured via environment variables. We will configure these environment variables in a sourced bash script, encode the environment file data, and store it in a secret which will be loaded at container runtime.
+
+The [New Relic Linux Server configuration page]
+(https://docs.newrelic.com/docs/servers/new-relic-servers-linux/installation-configuration/configuring-servers-linux) lists all the other settings for nrsysmond.
+
+To create an environment variable for a setting, prepend NRSYSMOND_ to its name. For example,
+
+```sh
+loglevel=debug
+```
+translates to
+
+```sh
+NRSYSMOND_loglevel=debug
+```
+
+Edit examples/newrelic/nrconfig.
+
+<!-- BEGIN MUNGE: EXAMPLE nrconfig -->
+
+```sh
+export NRSYSMOND_loglevel=debug
+export NRSYSMOND_license_key=REPLACE_WITH_REAL_KEY
+```
+
+[Download example](nrconfig?raw=true)
+<!-- END MUNGE: EXAMPLE nrconfig.yaml -->
+
+Be sure to edit the license key field and fill in your own NewRelic license key.
+
+Now, let's vendor the config into a secret.
+
+```sh
+cd examples/newrelic/
+./config-to-secret.sh
+```
+
+<!-- BEGIN MUNGE: EXAMPLE newrelic-secret-template.yaml -->
+
+```yaml
+apiVersion: v1
+kind: Secret
+metadata:
+  name: newrelic-keys
+type: Opaque
+data:
+  licensekey: {{license_key_data}}
+```
+
+[Download example](newrelic-secret-template.yaml?raw=true)
+<!-- END MUNGE: EXAMPLE newrelic-secret.yaml -->
+
+The script will encode the config file and write it to `newrelic-config.yaml`.
+
+Finally, submit the config to the cluster:
+
+```sh
+kubectl create -f examples/newrelic/newrelic-config.yaml
```

Comment:
I didn't do this because it's always annoying for me when I'm reading examples to remove the $. But, I'll add it if it's important.